src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -161,3 +178,144 @@ export const handleRedirect: RequestHandler<
+          formId,
+        },
+        error,
+      })
+    }
+    const { errorMessage, statusCode } = mapRouteError(error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  const form = formResult.value
+  const publicForm = form.getPublicView()
+  const { authType } = form
+  const isIntranetUser = FormService.checkIsIntranetFormAccess(
+    getRequestIp(req),
+    form,
+  )
+
+  switch (authType) {
+    case AuthType.NIL:
+      return res.json({ form: publicForm, isIntranetUser })
+    case AuthType.SP:
+    case AuthType.CP:
+      return SpcpFactory.getSpcpSession(authType, req.cookies)
+        .map((spcpSession) =>
+          res.json({
+            form: publicForm,
+            isIntranetUser,
+            spcpSession,
+          }),
+        )
+        .mapErr((error) => {
+          // Report only relevant errors - verification failed for user here
+          if (
+            error instanceof VerifyJwtError ||
+            error instanceof InvalidJwtError
+          ) {
+            logger.error({
+              message: 'Error getting public form',
+              meta: {
+                action: 'handleGetPublicForm',
+                ...createReqMeta(req),
+                formId,
+              },
+              error,
+            })
+          }
+          return res.json({ form: publicForm, isIntranetUser })
+        })
+    case AuthType.MyInfo: {
+      // Step 1. Fetch required data and fill the form based off data retrieved
+      return (
+        MyInfoFactory.fetchMyInfoData(form, req.cookies)
+          .andThen((myInfoData) => {
+            return MyInfoFactory.createFormWithMyInfoMeta(
+              form.toJSON().form_fields,
+              myInfoData,
+              form._id,
+            )
+          })
+          // Check if the user is signed in
+          .andThen(({ prefilledFields, spcpSession }) => {
+            return extractAndAssertMyInfoCookieValidity(req.cookies).map(
+              (myInfoCookie) => ({
+                prefilledFields,
+                spcpSession,
+                myInfoCookie,
+              }),
+            )
+          })
+          .map(({ myInfoCookie, prefilledFields, spcpSession }) => {
+            const updatedMyInfoCookie = {
+              ...myInfoCookie,
+              usedCount: myInfoCookie.usedCount + 1,
+            }
+            // Set the updated cookie accordingly and return the form back to the user
+            return res
+              .cookie(
+                MYINFO_COOKIE_NAME,
+                updatedMyInfoCookie,
+                MYINFO_COOKIE_OPTIONS,
+              )
+              .json({
+                spcpSession,
+                form: { ...publicForm, form_fields: prefilledFields },
+                isIntranetUser,
+              })
+          })
+          .mapErr((error) => {
+            // NOTE: If the user is not signed in or the access token has been used before, it is not an error.
```

Comment:
maybe can explain the "access token has been used before" case a bit more by specifying that it happens when user refreshes the page while logged in